apps/tangle-dapp/utils/entriesOf.ts

```diff
@@ -0,0 +1,10 @@
+const entriesOf = <Key extends string | number | symbol, Value>(
+  object: Record<Key, Value>,
+): [Key, Value][] => {
+  // Type cast is needed because `Object.entries()` returns `[string, Value][]`,
+  // in other words, it unfortunately downgrades the key type to `string` by
+  // default.
+  return Object.keys(object).map((key) => [key as Key, object[key as Key]]);
```

Comment:
I think the `key as Key` casting here not is strong enough, because JavaScript object keys are always coerced to a string.

From Typescript docs: https://www.typescriptlang.org/docs/handbook/2/keyof-types.html#the-keyof-type-operator

![CleanShot 2024-06-07 at 18 05 01@2x](https://github.com/webb-tools/webb-dapp/assets/60747384/89d47473-1a2e-47df-9238-82369233afac)

Like on this [example](https://www.typescriptlang.org/play/?#code/MYewdgzgLgBApmKAnAlnCB5AZjAvDAHgGk4BPeADygQBMIZpUwBzGAHxjAFcBbAIzhJ2DUvxAAbADQwAagENxXOAD4AFACgYMEHwBWcYFABcMAEoGQSGsTLT5ilZPUBKEwG0SpOwqUBdN754yjAA3powAPQRMAAqpAAOcDDActAwKPRgcHA0OTACKVwQSQAGGHoGUAB0CMhoEKrOJTBIcFBcSJAwJW6MKCzeDv6+JU5aUelg2lAAFoIwAO6WdNIosFxgWJbtYHLU4uQ0IAssSHK59LNJANZkMFAJSVAg3X0szXyk4RO5WHJc4mq4Va7U6MHK+kMVVupAaOkhUGcVR4cniqlUMOcQRgbhhMFSME80nhlVxdwJnl8vmcAG51ABfOnqUCQWDwvChcJaACMJgARNRoNy+WMtDAAEz8wVQcV8hnqZngNKtej4WqodDYVTwpGtGhcYBwdFyYDAaRkrwwAD6ADcfHBqdiTcAYABqGAw6QABlpCpZEAkcCq4hAzFUJXMEABxhgABIQir6SVaUA), if I use the key on some calculation, it will produce the incorrect result due to the type is incorrect.

Reply:
Thanks for catching this! I changed the generic constraint to accept `Key extends string` that way we still get the benefit of the key not downgrading to `string`, but remaining type `Key`, if and only if `Key extends string` already, which should cover the edge case you've mentioned.